test/proposal-genericscheme.spec.ts

```diff
@@ -33,30 +32,37 @@ describe('Proposal', () => {
     const states: IProposalState[] = []
     const lastState = (): IProposalState => states[states.length - 1]
 
-    const actionMockABI = arc.getABI(undefined, 'ActionMock', LATEST_ARC_VERSION)
+    const actionMockABI = arc.getABI({abiName: 'ActionMock', version: LATEST_ARC_VERSION})
 
     if(!arc.web3) throw new Error('Web3 provider not set')
 
-    const actionMock = new Contract(testAddresses.test.ActionMock.toString(), actionMockABI, arc.web3.getSigner())
     const callData = new ethers.utils.Interface(actionMockABI).functions.test2.encode([dao.id])
 
-    const schemes = await dao.schemes({ where: {name: 'GenericScheme' }}).pipe(first()).toPromise()
-    const genericScheme = schemes[0].coreState as ISchemeState
-    const proposal = await createAProposal(dao, {
+    const plugins = await dao.plugins({ where: {name: 'GenericScheme' }}).pipe(first()).toPromise() as GenericScheme[]
+    const genericScheme = plugins[0]
+
+    const tx = await genericScheme.createProposal({
+      dao: dao.id,
       callData,
-      scheme: genericScheme.address,
-      schemeToRegister: actionMock.address,
-      value: 0
-    })
+      value: 0,
+      proposalType: "GenericScheme",
```

Comment:
Given that we call "createProposal" on a genericSchme instance, we should be able to derive the "proposalType" from there. Also the next line, which requires a plugin: it's not clear to me why that is needed; `genericScheme` already the plugin here?

Reply:
I agree completely. Actually, I just realized that, since now each plugin has its own createProposal implementation, that parameter is not even necessary anymore. Except for the schemeRegistrar, which can create more than one proposal type. I removed proposalType from baseOptions and put it in SchemeRegistrar createproposal options.